tests/framework/integration/cluster.go

```diff
@@ -1410,32 +1410,41 @@ func (c *Cluster) Endpoints() []string {
 	return endpoints
 }
 
-func (c *Cluster) ClusterClient() (client *clientv3.Client, err error) {
-	if c.clusterClient == nil {
-		var endpoints []string
-		for _, m := range c.Members {
-			endpoints = append(endpoints, m.GrpcURL)
-		}
-		cfg := clientv3.Config{
-			Endpoints:          endpoints,
-			DialTimeout:        5 * time.Second,
-			DialOptions:        []grpc.DialOption{grpc.WithBlock()},
-			MaxCallSendMsgSize: c.Cfg.ClientMaxCallSendMsgSize,
-			MaxCallRecvMsgSize: c.Cfg.ClientMaxCallRecvMsgSize,
-		}
-		if c.Cfg.ClientTLS != nil {
-			tls, err := c.Cfg.ClientTLS.ClientConfig()
-			if err != nil {
-				return nil, err
-			}
-			cfg.TLS = tls
-		}
-		c.clusterClient, err = newClientV3(cfg)
+func (c *Cluster) ClusterClient(opts ...func(*clientv3.Config)) (client *clientv3.Client, err error) {
+	if c.clusterClient != nil && len(opts) == 0 {
```

Comment:
Adding ability to create a client from `opts` breaks assumptions needed to cache the client on `c.clusterClient`.

I think the original goal of caching client was removing need for user to call `.Close()` on client as this can be handled by Cluster.

If we allowing arguments for clients, I would prefer to remove caching altogether. Problem with closing client can be nicely done by adding it to `testing.T.Cleanup` as you nicely used in this PR above.

Reply:
Done in the next PR revision